colossalai/moe/layers.py

```diff
@@ -104,6 +105,7 @@ def __init__(
         if self.expert_parallel is not None:
             self.ep_group = get_ep_group(self.experts)
             self.ep_size = get_ep_size(self.experts)
+            self.ep_hierarchical_group = create_hierarchical_group(self.ep_group)
```

Comment:
Use an arg to use hierachical alltoall. Also need to add it in the config file and set_openmoe_arg function.

Reply:
c6d51c0 resolves this.